styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -180,13 +270,42 @@ private void updateStats(Map<String, Resource> resources,
 
   private boolean limitAndDequeue(Map<String, Resource> resources,
       Map<WorkflowId, Set<String>> workflowResourceReferences,
-      Map<String, Long> currentResourceUsage, InstanceState instance) {
+      Map<String, Long> currentResourceUsage, InstanceState instance,
+      CompletionStage<List<String>> missingDependenciesFuture) {
```

Comment:
This one should probably be more generic, maybe just a `String` with some message about why gate was closed.